docs/NETWORK-MODELS.md

```diff
@@ -0,0 +1,87 @@
+# Network Models
+
+What are the network models that EVE supports for Edge Container workloads, and how are they implemented?
+
+The first half of this document describes _what_ those network models are and how they impact Edge Container communications. The second half describes _how_ those models are implemented.
+
+## Models
+
+Fundamentally, each Edge Container may need to communicate with services outside itself. Those can be one or more of:
+
+* one or more Edge Containers on the same device
+* a communications network to which the device, on which the Edge Container is running, is attached via some i/o port, e.g. a NIC
+* one or more Edge Containers on one or more other devices
+* a network (VPC) in the public cloud
+* no networking at all
+
+In principle, an Edge Container definition does not define any of its methodologies, networks or technologies. It simply defines what it should be able to connect to. The Controller then is responsible for translating those networking permissions to implementations passed on to EVE.
+
+Put in other terms, the various complex methods of networking are the responsibility of the _Controller_ to define and expose to users. EVE itself, on the other hand, is responsible solely for providing basic networking building blocks that enable all of the necessary forms of communication, and which a Controller can meld together to provide the necessary networking services.
+
+## Definitions
+
+* `network`: a joining together of one or more endpoints that will be able to communicate with each other via packets
+* `endpoint`: a unique connection point to a network, includes Edge Containers and one or more communications media
+* `Edge Container`: a single running workload on a single device, which can be connected as an endpoint to 0, 1 or multiple networks simultaneously
+* `port`: an endpoint, which can be backed by a physical device, e.g. a NIC or a serial port
+
+## Responsibilities
+
+From the perspective of EVE on a single device, which is its entire scope of responsibility, EVE needs to be informed of the following:
+
+* that a network should exist on the device, and a representative name for it
+* what the network type is: L2 or L3 (see below)
+* the endpoints that are to be connected to that network (see below)
+
+Upon receiving these, EVE should either create a network that matches the specifications and connect endpoints, or update one if it already exists.
+
```

Comment:
Should we make it clear that the mechanism of "receiving" is via generic config?

Reply:
Yes, no problem. Updating.